pkg/controller/workload/job/job_controller.go

```diff
@@ -323,7 +341,7 @@ func (r *JobReconciler) ensureAtMostOneWorkload(ctx context.Context, job *batchv
 			// than one workload...
 			w = &workloads.Items[0]
 		}
-		if err := r.stopJob(ctx, w, job); err != nil {
+		if err := r.stopJob(ctx, w, job, "No matching QueuedWorkload"); err != nil {
```

Comment:
The messages should be a full sentence: Stopped Job because there was no matching QueuedWorkload

Reply:
Sounds redundant, the reason is "Stopped", why replicate that again if the context is clear?

Reply:
For consistency with how we do events in k8s

Reply:
messages in k8s seems quite adhoc to me :)